r-lib-error-1.0.md

```diff
@@ -0,0 +1,60 @@
+
+# Specification of r-lib errors, version 1.0
+
+Note: this is an experimental draft.
+
+Note: version 1.0 corresponds to the `rlang_error` class, as
+implemented in rlang 0.4.0.
+
+## Classes
+
+An r-lib error object always has at least classes `rlib_error`, `error`
+and `condition`. It may have other classes.
+
+## Members
+
+An `rlib_error` object has the following members:
+
+* `message`: `character(1)`, the error message.
+* `trace`: optional, an `rlib_trace` object, see below. 
+* `parent`: optional, pointer to a parent error, of class `error`. It
+  may also be `NULL`, which is the same as missing.
```

Comment:
Yeah, small different, it might be missing, or present, but `NULL`. Usually the does not matter, but since this is a semi-formal spec, I thought I would mention it.